src/types.ts

```diff
@@ -113,6 +117,22 @@ export interface PredefinedType extends BaseValueType {
   name: string;
 }
 
+export interface LiteralType extends BaseValueType {
+  kind: ValueTypeKind.literalType;
+  type:
+    | {
```

Comment:
This is identical to the definition of `BasicType`, why not simply use `BasicType`?

Reply:
I reuse it from parseLiteralNode and didn't change it

Reply:
Can you change it though?

Reply:
Changed LiteralType to UnionLiteralType and removed unnecessary content.